net-misc/bsdwhois/files/bsdwhois-1.43.2.1-clang16-build.patch

```diff
@@ -0,0 +1,34 @@
+Clang16 will not allow implicit function declarations and implicit integers etc.
+This patch overhauls the source code for modern C.
+
+Bug: https://bugs.gentoo.org/875029
+
+Signed-off-by: Pascal Jäger <pascal.jaeger@leimstift.de>
+
+--- a/strnstr.c
++++ b/strnstr.c
+@@ -46,10 +46,7 @@ static char sccsid[] = "@(#)strstr.c	8.1 (Berkeley) 6/4/93";
+  * first slen characters of s.
+  */
+ char *
+-strnstr(s, find, slen)
+-	const char *s;
+-	const char *find;
+-	size_t slen;
++strnstr(const char *s, const char *find, size_t slen)
+ {
+ 	char c, sc;
+ 	size_t len;
+--- a/whois.c
++++ b/whois.c
+@@ -103,6 +103,10 @@ static void s_asprintf(char **ret, const char *format, ...) __printflike(2, 3);
+ static void usage(void);
+ static void whois(const char *, const char *, int);
```

Comment:
This one is a bit suspicious.

If you take a look at `vasprintf(3)` (`man vasprintf`), you'll see we need `_GNU_SOURCE` for it, because these functions are not in POSIX (they're GNU extensions).

Try adding `#define _GNU_SOURCE` at the top of the file. But if it uses autotools, I prefer to use `AC_USE_SYSTEM_EXTENSIONS`. If upstream is a dead end and not using autotools, I often just `append-cppflags -D_GNU_SOURCE` in the ebuild. It's important to try set it consistently where possible.

Reply:
amIdoingthinright?